imagedephi/rules/ruleset.py

```diff
@@ -1,49 +1,88 @@
+    file_format = FileFormat.TIFF
+    metadata_rules: dict[int, MetadataTiffRule]
+
+    @staticmethod
+    def build_tiff_metadata_rule(
+        tag_name: str, method_spec: str | dict, rule_source: RuleSource
+    ) -> MetadataTiffRule:
+        rule_spec = {"tag_name": tag_name}
+        if isinstance(method_spec, str):
+            rule_spec["method"] = method_spec
+        else:
+            rule_spec = rule_spec | method_spec
+        return MetadataTiffRule.build(rule_spec, rule_source)
 
-    def get_metadata_tiff_rules(self) -> list[MetadataTiffRule]:
-        if FileFormat.TIFF in self.rules:
-            return [
-                rule for rule in self.rules[FileFormat.TIFF] if isinstance(rule, MetadataTiffRule)
-            ]
-        return []
+    def __init__(self, file_format_spec: dict, rule_source: RuleSource):
+        self.metadata_rules = {}
+        for tag_name, method_info in file_format_spec.get("metadata", {}).items():
+            tiff_tag = get_tiff_tag(tag_name)
+            self.metadata_rules[tiff_tag.value] = TiffFormatRules.build_tiff_metadata_rule(
+                tag_name, method_info, rule_source
+            )
 
-    def get_metadata_svs_rules(self) -> list[MetadataSvsRule]:
-        if FileFormat.SVS in self.rules:
-            return [
-                rule for rule in self.rules[FileFormat.SVS] if isinstance(rule, MetadataSvsRule)
-            ]
-        return []
 
+class SvsFormatRules(TiffFormatRules):
+    file_format = FileFormat.SVS
+    image_description_rules: dict[str, MetadataSvsRule]
 
-def _build_rule(file_format: FileFormat, rule_spec: dict, rule_source: RuleSource) -> Rule:
-    rule_type = RuleType[rule_spec["type"].upper()]
-    if file_format == FileFormat.TIFF:
-        if rule_type == RuleType.METADATA:
-            return MetadataTiffRule.build(rule_spec, rule_source)
-    if file_format == FileFormat.SVS:
-        if rule_type == RuleType.METADATA:
-            return MetadataSvsRule.build(rule_spec, rule_source)
+    @staticmethod
+    def build_svs_image_description_rule(
+        key_name: str, method_spec: str | dict, rule_source: RuleSource
+    ) -> MetadataSvsRule:
+        rule_spec = {"key_name": key_name}
+        if isinstance(method_spec, str):
+            rule_spec["method"] = method_spec
+        else:
+            rule_spec = rule_spec | method_spec
+        return MetadataSvsRule.build(rule_spec, rule_source)
+
+    def __init__(self, file_format_spec: dict, rule_source: RuleSource):
+        super().__init__(file_format_spec, rule_source)
+        self.image_description_rules = {}
+        for key_name, method_info in file_format_spec.get("image_description", {}).items():
+            self.image_description_rules[
+                key_name
+            ] = SvsFormatRules.build_svs_image_description_rule(key_name, method_info, rule_source)
+
+
+@dataclass
+class RuleSet:
+    name: str
+    description: str
+    tiff: TiffFormatRules
+    svs: SvsFormatRules
```

Comment:
This, combined with the new classes defined above, unify the structure of the RuleSet class and the shape of the rule set files.